xgcm/test/test_faceconnections.py

```diff
@@ -211,19 +229,59 @@ def test_diff_interp_connected_grid_x_to_y(ds, ds_face_connections_x_to_y):
     # TODO: checking all the other boundaries
 
 
-@pytest.mark.xfail(reason="connected grids not implemented for grid ufunc refactor yet")
+# TODO: Relaease the periodic test here once we unified the API with padding.
+@pytest.mark.parametrize(
+    "boundary", [pytest.param("periodic", marks=pytest.mark.xfail), "fill"]
+)
+def test_vector_connected_grid_x_to_y(ds, ds_face_connections_x_to_y, boundary):
+    # one face connection, rotated
+    grid = Grid(
+        ds,
+        face_connections=ds_face_connections_x_to_y,
+        boundary=boundary,
+        fill_value=1,
+        periodic=False,
+    )
+    # TODO: Remove the periodic once that is deprecated.
+    # ! Set boundary on grid, so it is applied to all axes.
+    # TODO: modify the non velocity tests too (after release)
+
+    # modify the values of the dataset, so we know what to expect from the output
+    # TODO: Maybe change this in the dataset definition?
+    u_modifier = xr.DataArray([-2, -1], dims="face")
+    v_modifier = xr.DataArray([1, 1], dims="face")
+    u = ds.u * 0 + u_modifier
+    v = ds.v * 0 + v_modifier
+
+    # no need to check for diff vs interp. They all go through the same dispatch
+    # v is the interesting variable here because it involves a sign change for this
+    # connection (see https://github.com/xgcm/xgcm/issues/410#issue-1098348557)
+    v_out = grid.interp({"Y": v}, "X", other_component={"X": u})
+    # the test case is set up in a way that all interpolated values for u should be 1
+    # if the face connection is done properly
+    np.testing.assert_allclose(v_out.data, 1)
```

Comment:
Why not use [`xarray.testing.assert_allclose`](https://xarray.pydata.org/en/v2022.03.0/generated/xarray.testing.assert_allclose.html)?

Reply:
Same as above

Reply:
Using `xr.testing.assert_allclose(v_out, 1)` gives me this rather unhelpful error:
```
raceback (most recent call last):
  File "<string>", line 1, in <module>
  File "[/Users/juliusbusecke/miniconda/envs/test_env_xgcm/lib/python3.9/site-packages/xarray/testing.py]()", line 34, in wrapper
    return func(*args, **kwargs)
  File "[/Users/juliusbusecke/miniconda/envs/test_env_xgcm/lib/python3.9/site-packages/xarray/testing.py]()", line 146, in assert_allclose
    assert type(a) == type(b)
AssertionError
```
Since the numpy version passes I assume that there is some issue with the coordinates? Not sure its worth digging into this here. The assertion is covering the part that matters here.

Reply:
It's complaining because it's pointing out that the DataArray is a different type to the 1, because the 1 is an int. I assume the numpy version of the assert must first attempt to turn the 1 into a numpy array.